pmd-core/src/main/java/net/sourceforge/pmd/rules/RuleBuilder.java

```diff
@@ -45,8 +47,9 @@
     private boolean isUsesMultifile;
     private boolean isUsesTyperesolution;
 
-    public RuleBuilder(String name, String clazz, String language) {
+    public RuleBuilder(String name, ResourceLoader resourceLoader, String clazz, String language) {
```

Comment:
In order to keep backwards compatible, we need to provide an overloaded constructor.
The one without a resource loader should initialize the resource loader as `new ResourceLoader()`, which uses the same classpath as before. We could mark this constructor already as deprecated, since it might load the rule implementation class from a wrong class loader.